ring/batch.go

```diff
@@ -134,49 +167,54 @@ func (b *batchTracker) record(itemTrackers []*itemTracker, err error) {
 	//
 	// The use of atomic increments here is needed as:
 	// * rpcsPending and rpcsFailed guarantee only a single goroutine will write to either channel
-	// * succeeded, failed4xx, failed5xx and remaining guarantee that the "return decision" is made atomically
+	// * succeeded, failedClient, failedServer and remaining guarantee that the "return decision" is made atomically
 	// avoiding race condition
-	for i := range itemTrackers {
+	for _, itemTracker := range itemTrackers {
+		if itemTracker.done.Load() {
+			continue
+		}
```

Comment:
what's the reason for this change?

Reply:
My idea was to set the new `done` attribute of an `itemTracker` to true as soon as that `itemTracker` reaches either its `minSuccess` or its `maxFailure`. Once it is achieved, we don't need to consider that `itemTracker` anymore.

Reply:
As discussed via slack, this will be done in a separate PR, and maybe in a different form. I have reverted this part of the PR.